static_frame/core/index_hierarchy.py

```diff
@@ -1282,92 +1947,100 @@ def _ufunc_binary_operator(self, *,
         from static_frame.core.series import Series
         from static_frame.core.frame import Frame
 
-        if self._recache:
-            self._update_array_cache()
-
         if isinstance(other, (Series, Frame)):
             raise ValueError('cannot use labelled container as an operand.')
 
         if operator.__name__ == 'matmul':
-            return matmul(self._blocks.values, other)
+            return matmul(self.values, other)
         elif operator.__name__ == 'rmatmul':
-            return matmul(other, self._blocks.values)
+            return matmul(other, self.values)
+
+        if self._recache:
+            self._update_array_cache() # pragma: no cover
 
         if isinstance(other, Index):
             other = other.values
         elif isinstance(other, IndexHierarchy):
             if other._recache:
-                other._update_array_cache()
+                other._update_array_cache() # pragma: no cover
```

Comment:
Why skip coverage of this scenario?

Reply:
See this response: https://github.com/InvestmentSystems/static-frame/pull/437#discussion_r831501537